source/agora/common/Deserializer.d

```diff
@@ -0,0 +1,142 @@
+/*******************************************************************************
+
+    Function definition and helper related to Deserialization
+
+    Copyright:
+        Copyright (c) 2019 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.common.Deserializer;
+
+import agora.common.Data;
+import agora.common.Block;
+import agora.common.crypto.Key;
+
+/// Type of delegate deserializeDg
+public alias DeserializeDg = ubyte[] delegate(size_t size) nothrow @safe;
+
+/*******************************************************************************
+
+    Deserialize a struct and return it as a Block.
+
+    Params:
+        T = Type of struct to deserialize
+        source = uByte[] source to be deserialized
+        dg  = Deserialization delegate, when this struct is a nested struct
+
+    Returns:
+        The deserialized `Block`
+
+*******************************************************************************/
+
+public Block deserializeBlock (scope ref ubyte [] data)
```

Comment:
```suggestion
public Block deserializeBlock (scope ubyte[] data)
```

I don't know if we can make this data `const`. Probably not.

Reply:
Applied the suggestion